### PR TITLE
disable EIGEN_FAST_MATH and use_fast_math

### LINKED ISSUE #13646: mkldnn batch norm maybe diff with mkl batch norm
The test machine is: E5-2620 v2, which only have AVX Instruction set.

When I inference resnet50 model: https://github.com/PaddlePaddle/Paddle/compare/develop...luotao1:resnet50?expand=1

The commit id is: https://github.com/PaddlePaddle/Paddle/commit/643b6faa0ced3304d5531dfa6a87a7fb39a54e0f

I run `ctest -R test_analyzer_resnet50 -V`.  following Test fails:
```
// Compare result of NativeConfig and AnalysisConfig
TEST(Analyzer_resnet50, compare) 
```
The error is that there is > 0.001 diff for all result data.
```
95: The difference between pdata_ref[j] and pdata[j] is 0.00164794921875, which exceeds 1e-3, where
95: pdata_ref[j] evaluates to 451.11993408203125,
95: pdata[j] evaluates to 451.1182861328125, and
95: 1e-3 evaluates to 0.001.
```
When `FLAGS_with_mkldnn=true`:
- NativeConfig: mkl
- AnalysisConfig: mkldnn. And to avoid `fc_fuse_pass`, I exclude this `fc_fuse_pass` as well. Thus, the only diff is between mkl op and mkldnn op.

However, as `test_analyzer_ocr` runs sucessfully. We compare the ops in [ocr model](https://github.com/PaddlePaddle/Paddle/issues/13574#issuecomment-424638054) and resnet50 model.  **The difference is batch_norm op**, which doesn't exist in ocr model.

Besides I also merge #13486 to see whether `conv bn fuse pass` runs sucessfully.
But both I set `cfg->_use_mkldnn = true` and `cfg->_use_mkldnn = false`, there is still diff.

Could you help see it?

### PR DESCRIPTION
**Pull request by luotao1**
fix #13646
This PR should be tested by CE at first.

### CODE REVIEW

**Comment by luotao1** [comment]
It pass the CE: http://ce.paddlepaddle.org:8080/viewLog.html?tab=buildLog&buildTypeId=build_PaddlePrDebugBuild2&buildId=1992&_focus=362#_state=362

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,6 +72,7 @@ option(WITH_INFERENCE    "Compile fluid inference library"              ON)
 option(WITH_INFERENCE_API_TEST   "Test fluid inference high-level api interface"  OFF)
 option(WITH_SYSTEM_BLAS   "Use system blas library"           OFF)
 option(PY_VERSION       "Compile PaddlePaddle with python3 support"     ${PY_VERSION})
+option(WITH_FAST_MATH   "Make use of fast math library"                 OFF)
```
Comment (Superjomn) [inline CMakeLists.txt:75]: Add some information to remind the users that " fast math might affect the precision to some extent."

**Review by panyx0718 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (luotao1) [reply CMakeLists.txt:75]: OK, I will add the information in next PR.
